commit 2c810ccdbac434ae38f4ec5331d3f047dc90f98a
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Thu Apr 29 00:53:29 2010 -0700

    iwlwifi: rework broadcast station management
    
    Currently, the broadcast station is managed along
    with the interface type, rather than always being
    present. That leads to a bug with injection -- it
    is currently not possible to inject frames when
    the only virtual interface is a monitor, because
    in that the required broadcast station is missing.
    
    Additionally, allocating and deallocating the
    broadcast station's LQ all the time is wasteful,
    and the code to support this is fairly complex.
    
    So this changes completely the way we manage the
    broadcast station. Rather than manage it along
    with any interface, we now allocate it when we
    bring the device up, and remove it again when we
    bring the device down. When we bring the device
    up, we don't immediately program the broadcast
    station into it, instead we just mark it active
    and rely on the next restore cycle to upload it
    to the device. This works because an unassociated
    RXON is always required at least once to set up
    device parameters, which implies a reprogramming
    of stations into the device.
    
    As we now manage all stations properly, there no
    longer is a need for forcing a clearing of them
    via iwl_clear_ucode_stations(), which can become
    a lot simpler.
    
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>
    Signed-off-by: Reinette Chatre <reinette.chatre@intel.com>

diff --git a/drivers/net/wireless/iwlwifi/iwl-1000.c b/drivers/net/wireless/iwlwifi/iwl-1000.c
index a2f7cbcc2f7c..6be2992f8f21 100644
--- a/drivers/net/wireless/iwlwifi/iwl-1000.c
+++ b/drivers/net/wireless/iwlwifi/iwl-1000.c
@@ -212,7 +212,6 @@ static struct iwl_lib_ops iwl1000_lib = {
 		.temperature = iwlagn_temperature,
 		.set_ct_kill = iwl1000_set_ct_threshold,
 	 },
-	.add_bcast_station = iwl_add_bcast_station,
 	.manage_ibss_station = iwlagn_manage_ibss_station,
 	.debugfs_ops = {
 		.rx_stats_read = iwl_ucode_rx_stats_read,